compiler.js

```diff
@@ -392,6 +397,17 @@
             return section;
         };
 
+        this.addJS = function (js, story, section, passage, isFirst, inputFilename, lineCount) {
+            if (!passage) {
+                section = this.ensureSectionExists(story, section, isFirst, inputFilename, lineCount);
```

Comment:
I can see you're following the pattern established here by other functions such as `addAttribute`, but I'm not sure it makes sense here. `@deactivate` only makes sense in a passage that is accessed via a link, so there's no need to handle the case where it's added to a section at all, let alone an initial unnamed section in a file (which is what `ensureSectionExists` does).